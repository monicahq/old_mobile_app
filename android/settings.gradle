rootProject.name = 'Monica'
include ':react-native-vector-icons'
project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
include ':react-native-splash-screen'
project(':react-native-splash-screen').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-splash-screen/android')
include ':react-native-device-info'
project(':react-native-device-info').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-device-info/android')
include ':react-native-code-push'
project(':react-native-code-push').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-code-push/android/app')
include ':appcenter-crashes'
project(':appcenter-crashes').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter-crashes/android')
include ':appcenter-analytics'
project(':appcenter-analytics').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter-analytics/android')
include ':appcenter'
project(':appcenter').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter/android')

include ':app'
