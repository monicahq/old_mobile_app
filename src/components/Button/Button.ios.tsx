import React, {PureComponent} from 'react';
import {ActivityIndicator, Text, TouchableOpacity} from 'react-native';
import {IButtonProps} from './Button.props';
import {styles} from './Button.styles';

export class Button extends PureComponent<IButtonProps> {
  public static defaultProps = {
    disabled: false,
    loading: false,
  };

  public getTitle() {
    const {title, children, loadingTitle, loading} = this.props;
    if (loading && loadingTitle) {
      return loadingTitle;
    }
    return title || children || '';
  }

  public render() {
    const {onPress, disabled, loading, id} = this.props;

    return (
      <TouchableOpacity
        testID={id}
        style={[styles.button, disabled && styles.buttonDisabled]}
        onPress={onPress}
        disabled={disabled}
        activeOpacity={0.4}
      >
        <Text style={styles.buttonText}>{this.getTitle()}</Text>
        {loading && (
          <ActivityIndicator color="#ffffff" style={styles.activityIndicator} />
        )}
      </TouchableOpacity>
    );
  }
}
