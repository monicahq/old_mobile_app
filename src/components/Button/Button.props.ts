export interface IButtonProps {
  id?: string;
  onPress: (...args) => any;
  children?: string;
  loading?: boolean;
  title?: string;
  disabled?: boolean;
  loadingTitle?: string;
}
