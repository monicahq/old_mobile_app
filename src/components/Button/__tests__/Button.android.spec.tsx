import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Button} from '../Button.android';

describe('Components', () => {
  describe('Button Android', () => {
    const defaultOnPress = {
      onPress: () => {
        return;
      },
    };

    it('should renders correctly', () => {
      expect(
        toJson(shallow(<Button {...defaultOnPress}>Test</Button>))
      ).toMatchSnapshot();
      expect(
        toJson(shallow(<Button {...defaultOnPress} title="qdwdqw" />))
      ).toMatchSnapshot();
      expect(
        toJson(
          shallow(
            <Button {...defaultOnPress} loading={true} loadingTitle="qdwdqw" />
          )
        )
      ).toMatchSnapshot();
      expect(
        toJson(
          shallow(<Button {...defaultOnPress} title="qdwdqw" disabled={true} />)
        )
      ).toMatchSnapshot();
    });

    // TODO: uncomment this
    // it('should contain the title uppercased passed into parameter', () => {
    //   const button1 = shallow(<Button {...defaultOnPress} title="Test" />);
    //   expect(button1.find('Text').text()).toEqual('TEST'); // eslint-disable-line

    //   const button2 = shallow(<Button {...defaultOnPress} title="Test2" />);
    //   expect(button2.find('Text').text()).toEqual('TEST2'); // eslint-disable-line
    // });

    it('should call onPress when taping the button', () => {
      const onPress = jest.fn();
      const button = shallow(<Button onPress={onPress} />);
      button.prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
