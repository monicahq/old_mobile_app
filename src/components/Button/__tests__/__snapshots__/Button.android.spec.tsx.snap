// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Button Android should renders correctly 1`] = `
<DummyTouchableNativeFeedback
  disabled={false}
  onPress={[Function]}
>
  <ForwardRef(View)
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(43, 187, 72)",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
        false,
      ]
    }
  >
    <ForwardRef(Text)
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      TEST
    </ForwardRef(Text)>
  </ForwardRef(View)>
</DummyTouchableNativeFeedback>
`;

exports[`Components Button Android should renders correctly 2`] = `
<DummyTouchableNativeFeedback
  disabled={false}
  onPress={[Function]}
>
  <ForwardRef(View)
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(43, 187, 72)",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
        false,
      ]
    }
  >
    <ForwardRef(Text)
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      QDWDQW
    </ForwardRef(Text)>
  </ForwardRef(View)>
</DummyTouchableNativeFeedback>
`;

exports[`Components Button Android should renders correctly 3`] = `
<DummyTouchableNativeFeedback
  disabled={false}
  onPress={[Function]}
>
  <ForwardRef(View)
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(43, 187, 72)",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
        false,
      ]
    }
  >
    <ForwardRef(Text)
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      QDWDQW
    </ForwardRef(Text)>
    <ForwardRef(ActivityIndicator)
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
  </ForwardRef(View)>
</DummyTouchableNativeFeedback>
`;

exports[`Components Button Android should renders correctly 4`] = `
<DummyTouchableNativeFeedback
  disabled={true}
  onPress={[Function]}
>
  <ForwardRef(View)
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(43, 187, 72)",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
        Object {
          "backgroundColor": "rgba(43, 187, 72, 0.4)",
        },
      ]
    }
  >
    <ForwardRef(Text)
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      QDWDQW
    </ForwardRef(Text)>
  </ForwardRef(View)>
</DummyTouchableNativeFeedback>
`;
